api/db-cdr/prep-tables/make-prep-survey.py

```diff
@@ -0,0 +1,455 @@
+    files_exist(date)
+
+    # setup csv directory to hold output csv files
+    if os.path.exists(home_dir):
+        shutil.rmtree(home_dir)
+    os.mkdir(home_dir)
+
+    for name in surveys:
+        rows = read_csv(name + '_' + date + '.csv')
+        print("Reading & Parsing... " + name)
+
+        # open your file writers for this survey and write headers
+        controlled_writer, registered_writer, all_writer = \
+            open_writers_with_headers(name)
+
+        # made these global in order to add rows in selected output files
+        global id_controlled
+        global id_registered
+        global id_all
+
+        # will use when the row_number = 1 to add in as the first row
+        row_number = 0
+        id_controlled = 1
+        id_registered = 1
+        id_all = 1
+
+        # initializing these variables in the case that the topic is null
+        topic_id_all = 0
+        topic_id_controlled = 0
+        topic_id_register = 0
+
+        for row in rows:
+
+            if not_row_to_skip(row):
+                field_annotation, field_annotation_array, question_label, question_code, \
+                text_validation, choices, section_header, min_value, \
+                max_value = parse_row_columns(row)
+                row_number += 1
+
+                item_flags = {}
+                for item in annotation_flags:
+                    if item in field_annotation:
+                        item_flags[item] = 1
+                    else:
+                        item_flags[item] = 0
+
+                long_code_to_short_code = {}
+                for item in field_annotation_array:
+                    if item not in annotation_flags:
+                        split_codes = item.split('=')
+                        if "CONTROLLED_ANSWER_SUPPRESSED" not in split_codes and \
+                                "REGISTERED_ANSWER_SUPPRESSED" not in split_codes:
+                            # in the case that it doesn't have a code ex: Launched 5/30/2017 (PTSC) & 12/10/2019 (CE).
+                            if len(split_codes) > 1:
+                                long_code_to_short_code[split_codes[0].strip().lower()] = \
+                                    split_codes[1].strip().lower()
+
+                answer_suppression = {}
+                for item in field_annotation_array:
+                    split = item.split('=')
+                    if "CONTROLLED_ANSWER_SUPPRESSED" in split:
+                        answer_suppression.setdefault("CONTROLLED_ANSWER_SUPPRESSED", [])
+                        answer_suppression["CONTROLLED_ANSWER_SUPPRESSED"].append(split[1].lower())
+
+                    elif "REGISTERED_ANSWER_SUPPRESSED" in split:
+                        answer_suppression.setdefault("REGISTERED_ANSWER_SUPPRESSED", [])
+                        answer_suppression["REGISTERED_ANSWER_SUPPRESSED"].append(split[1].lower())
+
+                # for the first row of survey
+                if row_number == 1:
```

Comment:
move this section to the top of the method since it breaks out of the loop quickly